ERCS/erc-7769.md

```diff
@@ -0,0 +1,546 @@
+
+Sets the reputation of given addresses.
+
+**Parameters:**
+
+* An array of reputation entries to add/replace, with the fields:
+
+  * `address` - the address to set the reputation for
+  * `opsSeen` - number of times a user operations with that entity was seen and added to the mempool
+  * `opsIncluded` - number of times user operations that use this entity was included on-chain
+
+* **EntryPoint** the entrypoint used by `eth_sendUserOperation`
+
+```json=
+# Request
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "method": "debug_bundler_setReputation",
+  "params": [
+    [
+      {
+        "address": "0x7A0A0d159218E6a2f407B99173A2b12A6DDfC2a6",
+        "opsSeen": "0x14",
+        "opsIncluded": "0x0D"
+      }
+    ],
+    "0x1306b01bC3e4AD202612D3843387e94737673F53"
+  ]
+}
+
+# Response
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": "ok"
+}
+```
+
+
+#### * debug_bundler_dumpReputation
+
+Returns the reputation data of all observed addresses.
+Returns an array of reputation objects, each with the fields described above in `debug_bundler_setReputation`.
+
+**Parameters:**
+
+* **EntryPoint** the entrypoint used by `eth_sendUserOperation`
+
+**Return value:**
+
+An array of reputation entries with the fields:
+
+* `address` - the address to set the reputation for
+* `opsSeen` - number of times a user operations with that entity was seen and added to the mempool
+* `opsIncluded` - number of times user operation that use this entity was included on-chain
+* `status` - (string) The status of the address in the bundler (`'ok'` | `'throttled'` | `'banned'`)
+
+```json=
+# Request
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "method": "debug_bundler_dumpReputation",
+  "params": ["0x1306b01bC3e4AD202612D3843387e94737673F53"]
+}
+
+# Response
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": [
+    { "address": "0x7A0A0d159218E6a2f407B99173A2b12A6DDfC2a6",
+      "opsSeen": "0x14",
+      "opsIncluded": "0x13",
+      "status": "ok"
+    }
+  ]
+}
+```
+
+#### * debug_bundler_addUserOps
+
+Inject `UserOperation` objects array into the mempool.
+Assume the given `UserOperation` objects all pass validation without actually validating them,
+and accept them directly into the mempool.
+
+**Parameters:**
+
+* An array of `UserOperation` objects
+
+```json=
+# Request
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "method": "debug_bundler_addUserOps",
+  "params": [
+    [
+      { sender: "0xa...", ... },
+      { sender: "0xb...", ... }
+    ]
+  ]
+}
+
+# Response
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": "ok"
+}
+```
+
+## Rationale 
+
+* explicit debug functions: bundlers are required to provide a set of debug functions, so that the "bundler specification test suite" can be used to verify its adherance to the spec.
+
+## Backwards Compatibility
+
+This proposal defines a new JSON-RPC API standard that does not pose any backwards compatibility challenges.
+
+## Reference Implementation
+
+See `https://github.com/eth-infinitism/bundler`
```

Comment:
```suggestion
```

Please either remove this section or move your reference implementation to the `assets/` directory.